File: src/main/java/komu/demodel/utils/ExtensionFileFilter.java
/*
 * Copyright (C) 2006 Juha Komulainen. All rights reserved.
 */
package komu.demodel.utils;

import java.io.File;
import java.io.FileFilter;

public final class ExtensionFileFilter implements FileFilter {

    private final String suffix;
    
    public ExtensionFileFilter(String extension) {
        this.suffix = "." + extension;
    }
    
    public boolean accept(File file) {
        return file.isFile() && file.getName().endsWith(suffix);
    }
}

File: src/main/scala/komu/demodel/utils/ExtensionFileFilter.scala
/*
 * Copyright (C) 2006-2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.utils

import java.io.{ File, FileFilter }

final class ExtensionFileFilter(extension: String) extends FileFilter {
  def accept(file: File) = file.isFile() && file.getName().endsWith("." + extension)
}
